src/resources/components/message/Message.model.js

```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose')
+
+const Schema = mongoose.Schema // extract Schema from mongoose.Schema
+
+const messageSchema = new Schema(
+  {
+    message: {
+      text: { type: String, required: true }
+    },
+
+    users: [
+      {
+        user: {
+          type: Schema.Types.ObjectId,
+          ref: 'User',
+          required: true
+        },
+        sender: {
+          type: Schema.Types.ObjectId,
+          ref: 'User',
+          required: true
+        }
+      }
+    ],
+
+    read: { type: Date }
```

Comment:
What does this 'read' property used for?

Reply:
That 'read' property is to show when the message was read by the receiver.

Reply:
That's a great idea! so a 'sender' property is needed inside 'message:' object. This way we can identify who is the sender and who is the receiver.
```
message: {
  text: { type: String, required: true },
  sender: { type: String, required: true }
}
```